traitsui/testing/tester/wx/implementation/enum_editor.py

```diff
@@ -0,0 +1,96 @@
+#  Copyright (c) 2005-2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+
+from traitsui.wx.enum_editor import (
+    ListEditor,
+    RadioEditor,
+    SimpleEditor,
+)
+from traitsui.testing.tester import command, locator
+from traitsui.testing.tester.wx import helpers
+
+
+class _IndexedEditor:
+    """ Wrapper for Editor and index """
+
+    target_class = None
+    handlers = []
+    solvers = []
+
+    def __init__(self, target, index):
+        self.target = target
+        self.index = index
+
+    @classmethod
+    def register(cls, registry):
+        registry.register_solver(
+            target_class=cls.target_class,
+            locator_class=locator.Index,
+            solver=lambda wrapper, location: 
+                cls(target=wrapper.target, index=location.index),
+        )
+        for interaction_class, handler in cls.handlers:
+            registry.register_handler(
+                target_class=cls,
+                interaction_class=interaction_class,
+                handler=handler
+            )
+        for location_class, solver in cls.solvers:
+            registry.register_solver(
+                target_class=cls,
+                interaction_class=location_class,
+                solver=solver,
+            )
```

Comment:
I realized the code for each of the classes here was nearly identical so I refactored to this.  This class has nothing to do with Enum editor specifically though, and should be pulled out into a separate module.  It could easily be used by CheckListEditor as well for example.  I am not exactly sure where to put it though.  It doesn't belong in common_ui_targets as it is not intended to be a target.  It is meant to be subclassed.  In fact it is also not even toolkit specific and could easily be used by qt Enum editor and Check list editor as well.

@kitchoi what do you think about having this as a skeleton class?  Do you think it is beneficial/if so where do you think it should be placed?